### PR TITLE
Fix onclose timeout

### PR DESCRIPTION
**Pull request by ncrohn**
This is a fix for 1206. The onclose on the socket is getting erroneously called. This fixes the issue of buffer taking precedence.

I don't think this is an ultimate solution, for that we need to plan on removing socket.io. This will give us time to not rush that though.

### CODE REVIEW

**Comment by mjpizz** [comment]
oh rly? Fun. I say :ship: to console-dev and have the team use it next week (since it's a little riskier when we do things like edit core socketio functionality)

**Comment by bdimcheff** [comment]
it's on console.dev, so yes

File: dist/socket.io.js [hunk 1]
```diff
@@ -2399,8 +2405,8 @@ var io = ('undefined' === typeof module ? {} : module.exports);
    * @api public
    */
 
-  // Do to a bug in the current IDevices browser, we need to wrap the send in a 
-  // setTimeout, when they resume from sleeping the browser will crash if 
+  // Do to a bug in the current IDevices browser, we need to wrap the send in a
```
Comment (aawilson) [inline dist/socket.io.js:2408]: Due, not do

Comment (bdimcheff) [inline dist/socket.io.js:2408]: this was a whitespace fix to a comment that comes from upstream

**Comment by mjpizz** [comment]
is this branch being used? should we merge into our master?

**Comment by pwhisenhunt** [comment]
I don't believe it is, I'm pretty sure I branched my fix.reconnect off of it. Currently we're hitting the github.com/olark/socket.io.git#fix.reconnection branch for opconsole, so this can more then likely just be removed.

**Comment by mjpizz** [comment]
this was merged into #4

**Pull request closed without merging**